src/server/routes/api/classrooms/index.js

```diff
@@ -240,7 +271,7 @@ router.post("/:classroomId/ask", async (req, res) => {
       { students: req.user._id }
     ],
   }, {
-    $push: {
+    $addToSet: {
```

Comment:
I think this should still be $push. $addToSet doesn't guarantee it will be appended to the end and with a complex object that can't be tested for equality it will never fail to add an item.